include/seastar/http/common.hh

```diff
@@ -45,6 +46,8 @@ namespace httpd {
 SEASTAR_MODULE_EXPORT_BEGIN
 
 class parameters {
+    // Note: the path matcher adds parameters with the '/' prefix into the params map (eg. "/param1"), and some getters
```

Comment:
I'm afraid I don't understand this patch at all. As I also noticed and fixed in my own "version" of your patch, https://github.com/scylladb/seastar/pull/2125 (I didn't see your PR when I wrote mine), we're indeed missing the decoding of the path. So my patch just added the missing decoding call, and that's it!
Why did you patch become so much more complicated?
If your PR adds new functionality, please put it in a separate PR or at least a separate patch in the same PR, with justification for the new functionality. This patch should have just had the call to decode_path() in the place (actually two places) it was previously missing.
Or maybe I just misunderstood this patch?